### DISCUSSION TITLE
Pretty-print (format) errors / stacktraces

### DISCUSSION THREAD

**Discussion by alpox** [babashka/nbb#296] [Q&A] [answered]
Is there a way to print a nice error message/stacktrace out of errors occurring? The error output I’m seeing looks fine when I execute the script with nbb but not if I’m using the example script from https://github.com/babashka/nbb/blob/main/doc/publish/README.md (index.mjs).
In addition I could not find a way to print a good looking error message from within clojurescript either. I’m getting sci errors that look alike:
```
#error {:message "exec: fatal: Failed to resolve 'true' as a valid ref.\n", :data {:type :sci/error, :line 74, :column 9, :message "exec: fatal: Failed to resolve 'true' as a valid ref.\n" ....
```
Is there a way to format those? I found `sci/format-stacktrace` but it doesn’t seem to be available within `nbb`

**Comment by borkdude** [comment, marked as answer]
I think we could expose this function https://github.com/babashka/nbb/blob/de2627563d112de292c4a09be5d35c78d021eb3c/src/nbb/error.cljs#L80 to the JS API so you can try/catch nbb evaluations and then pass the exception to this function to get the nice error report

**Reply by alpox** [reply]
That sounds like exactly what I need. I already tracked down that function before but was not able to access it.

**Reply by alpox** [reply]
In addition it could be useful to have that error-handler available within the sci execution - e.g. now I have a try/catch handler within sci where I receive the above error in the catch clause. It would be great if - there too - I could print a nicely formatted error.

**Reply by borkdude** [reply]
Fixed with fd8329db02de6260ed3f98d72a8b215225c7646b

**Reply by borkdude** [reply]
Docs here: https://github.com/babashka/nbb#printing-errors

**Reply by alpox** [reply]
This is big. Thanks a lot!
